build-logic-commons/code-quality/src/main/kotlin/gradlebuild/ci/PublishingCiArtifacts.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.ci
+
+import java.io.File
+
+/**
+ * Indicates a task may publish some artifacts for CI.
+ * For example, `DistributionTest` publishes the test daemon registry directory,
+ * `PerformanceTest` publishes performance reports.
+ */
+interface PublishingCiArtifacts {
```

Comment:
If we make the rules to extract reports extensible, we don't need this interface any more. Though I am also not sure what is better here, the interface seems like a good solution as well for our custom tasks.

Though if we need to use an extension for the plugin tasks anyway, maybe it is better to use the extension for our custom tasks as well for consistency.